src/service/alert.service.ts

```diff
@@ -2,11 +2,37 @@ import { Injectable, Sanitizer, SecurityContext } from '@angular/core';
 import { TranslateService } from 'ng2-translate';
 import { ConfigHelper } from '../helper';
 
+export type AlertType =  'success' | 'danger' | 'warning' | 'info';
+
+export class Alert {
+    id: number;
+    type: AlertType;
+    msg: string;
+    params?: any;
+    timeout: number;
+    toast: boolean;
+    position: string;
+    scoped: boolean;
+    close: (alerts: Alert[]) => void;
+}
+
+export class AlertOptions {
```

Comment:
Yes what I found is that you add types to existing JS code, it's always makes you re-think this code.
I wanted to keep them as 2 classes as it seemed that some some fileds of AlertOptions were optional when they were not in Alert.